src/ipc/ipc-common/IPCProxyBinderBase.cpp

```diff
@@ -47,7 +47,7 @@ void IPCProxyBinderBase::connectToServer()
 
 bool IPCProxyBinderBase::isReadyToConnect() const
 {
-    return m_componentCompleted && enabled() && !objectPath().isEmpty();
+      return m_componentCompleted && enabled() && (!objectPath().isNull())?(!objectPath().isEmpty()):(false);
```

Comment:
this is hard to read. Please split this into pieces, e.g. extract object path condition. Also please remember that C++ works from left to right, so you do not have to add there explict false, something like this is enough: ... && !objectPath().isNull() && !objectPath.isEmpty()...

Reply:
Done